test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -291,35 +301,21 @@ public async Task DiscoverAsync_withValidDotNetToolsPluginLinux_FindsThePlugin()
                 File.WriteAllText(myPlugin, string.Empty);
                 Mock<IEnvironmentVariableReader> environmentalVariableReader = new Mock<IEnvironmentVariableReader>();
                 environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.PluginPaths)).Returns(pluginPath);
-
-                using (var process = new Process())
+                SetFileExecutable(myPlugin, true);
+                using (var discoverer = new PluginDiscoverer(environmentalVariableReader.Object))
                 {
-                    // Use a shell command to make the file executable
-                    process.StartInfo.FileName = "/bin/bash";
-                    process.StartInfo.Arguments = $"-c \"chmod +x {myPlugin}\"";
-                    process.StartInfo.UseShellExecute = false;
-                    process.StartInfo.RedirectStandardOutput = true;
-                    process.Start();
-                    process.WaitForExit();
-
-                    if (process.ExitCode == 0)
-                    {
-                        using (var discoverer = new PluginDiscoverer(environmentalVariableReader.Object))
-                        {
-                            // Act
-                            var result = await discoverer.DiscoverAsync(CancellationToken.None);
-
-                            // Assert
-                            var discovered = false;
+                    // Act
+                    var result = await discoverer.DiscoverAsync(CancellationToken.None);
 
-                            foreach (PluginDiscoveryResult discoveryResult in result)
-                            {
-                                if (myPlugin == discoveryResult.PluginFile.Path) discovered = true;
-                            }
+                    // Assert
+                    var discovered = false;
 
-                            Assert.True(discovered);
-                        }
+                    foreach (PluginDiscoveryResult discoveryResult in result)
+                    {
+                        if (myPlugin == discoveryResult.PluginFile.Path) discovered = true;
                     }
+
+                    Assert.True(discovered);
```

Comment:
This is not asserting for each of the `result`s. Is that intentional? The assertion will only pass if there is one that passes this condition. If that is the intention of this code, then you can simply break out of the for loop as soon as you find that one matches. Perhaps something a little more concise would be better as well. I like to avoid for loops and if statements in test code as much as possible.

Would something like this work? (please check for correctness)

 ```suggestion
                    Assert.Contains(result, discoveryResult => myPlugin == discoveryResult.PluginFile.Path);
```